### ISSUE TITLE
InheritanceDiagram Results Misleading in Inheritance Chains containing Base Classes

### ISSUE THREAD

**Issue by jpmedley** [mdn/yari#3562]
For example, the Web Audio spec has a thing called `BaseAudioContext` which inherits from `EventTarget`. At least a dozen interfaces such as `AudioContext`, subclass this event. The inheritance diagram macro for `AudioContext`, produces this:

![image](https://user-images.githubusercontent.com/8204171/163823652-ada728fb-2061-4d56-9cb9-5cdb69a14574.png)

`BaseAudioContext` and `EventTarget` are currently documented; however, I can't get instances of it them in JavaScript and they have no members. These pages could be removed, but we would still see them in the inheritance chain. And we would still need to document their members.

Basically, if we can't copy these interfaces' members to derived classes, we need idioms, both in the inheritance diagrams and the pages they link to, that clearly define these as non-instantiable.

**Comment by schalkneethling** [comment]
@Elchi3 I wonder, did your recent rewrite resolve this issue?

**Comment by jpmedley** [comment]
No, it didn't. I just realized that my example was bad. I rewrote the description with a more appropriate inheritance chain and add a statement clarifying what I think should be done.

**Comment by Elchi3** [comment]
Joe is right. `BaseAudioContext` is in fact an abstract / non-constructible class. A discussion on how to properly document such API classes that work with inheritance or abstract parents is https://github.com/mdn/content/discussions/5798. I guess whatever comes out of that discussion will then inform if or how to change the inheritance diagram.

**Comment by caugner** [comment]
Closing as won't fix, as this issue is not clearly actionable.
